crates/sui-framework/sources/coin.move

```diff
@@ -27,6 +31,25 @@ module sui::coin {
         balance: Balance<T>
     }
 
+    /// Each Coin<T> will have a unique instance of CoinMetadata<T> that
+    /// stores the metadata for this coin type.
+    struct CoinMetadata<phantom T> has key, store {
+        id: UID,
+        /// Number of decimal places the coin uses.
+        /// A coin with `value ` N and `decimals` D should be shown as N / 10^D
+        /// E.g., a coin with `value` 7002 and decimals 3 should be displayed as 7.002
+        /// This is metadata for display usage only.
+        decimals: u8,
+        /// Name for the token
+        name: string::String,
+        /// Symbol for the token
+        symbol: ascii::String,
```

Comment:
I think by convention, `symbol` will always be part of `T`. Do we want to duplicate it here?

Reply:
IMO yes, as mentioned in the "Summary -> Consideration" section, although we already have the convention of using the T as in Coin<T>as the Coin Symbol, the symbol field in CoinMetadata will be useful in case people wants to change the symbol dynamically, e.g., "FB" -> "META" without upgrading the Move module